tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -332,6 +525,25 @@ compressed_batch_make_next_tuple(DecompressChunkState *chunk_state,
 	}
 }
 
+static bool
+compressed_batch_vector_qual(DecompressChunkState *chunk_state, DecompressBatchState *batch_state)
+{
+	Assert(batch_state->total_batch_rows > 0);
+	Assert(batch_state->next_batch_row < batch_state->total_batch_rows);
+
+	const int output_row = batch_state->next_batch_row;
+	const size_t arrow_row = unlikely(chunk_state->reverse) ?
```

Comment:
Could you add a comment on why this direction is more unlikely?

Reply:
The idea is to tell the compiler that this branch is well predictable so that it doesn't generate a conditional move. IIRC it had a performance effect in some of my benchmarks. I'll add a comment.

Reply:
I agree that this is very predictable and does not change during query execution. But if you declare it as `unlikely` and the expression is true, wouldn't we see a huge performance penalty here because the branch assumption is wrong?

Reply:
IRC if the branch is well predicted as either taken or not taken, it has similar cost, so it doesn't matter much if it's `likely` or `unlikely`, just that it's a branch and not a conditional move. I'll check, maybe I can just remove it, it looks too low level.

Reply:
I removed it, apparently it doesn't influence the performance in this case: https://grafana.ops.savannah-dev.timescale.com/d/fasYic_4z/compare-akuzm?orgId=1&var-branch=All&var-run1=2684&var-run2=2688&var-threshold=0.05&var-use_historical_thresholds=false&chunkNotFound